src/hal/src/acceleration_structure.rs

```diff
@@ -0,0 +1,224 @@
+    Build,
+    /// specifies that the destination acceleration structure will be built using data in a source acceleration structure, updated by the specified geometries.
+    /// Note there's constraints on update: https://microsoft.github.io/DirectX-Specs/d3d/Raytracing.html#acceleration-structure-update-constraints
+    Update,
+}
+
+/// A description of the geometry data needed to populate an acceleration structure.
+///
+/// TODO: there's something here that smells w/ what fields are needed to get the required build size vs what fields are needed to actually build. Also, the top/bottom levels having different requirements on which fields are valid.
+// TODO: I don't like that this is reused for the "get build sizes" and "do actual build" with complex rules on what fields are ignored when. Perhaps we could use DX12's model for `D3D12_BUILD_RAYTRACING_ACCELERATION_STRUCTURE_DESC` and `D3D12_BUILD_RAYTRACING_ACCELERATION_STRUCTURE_INPUTS` and move at least the src/dst fields onto the actual "build acceleration structure" command.?
+#[derive(Debug)]
+pub struct AccelerationStructureGeometryDesc<'a, B: Backend> {
+    pub flags: AccelerationStructureFlags,
+    /// The type of acceleration structure to build.
+    pub ty: AccelerationStructureType,
+    pub mode: AccelerationStructureCreateMode,
+
+    pub src: Option<B::AccelerationStructure>,
+    pub dst: Option<B::AccelerationStructure>,
+
+    // TODO: We could enforce the following the type system?
+    // - in both vulkan (via `VUID-VkAccelerationStructureBuildGeometryInfoKHR-type-03792`) and DX12 (via type system), all of the structs here must be the same variant.
+    // - blas must be triangles or aabbs, tlas must be instances
+    pub geometries: &'a [AccelerationStructureGeometry<'a, B>],
+    // Both APIs support "array" and "array of pointers", presumably to allow for cheap reuse of the geometry descriptors.
+    // pgeometries: &'a [&'a GeometryDesc<'a, B>],
+    pub scratch: Option<(&'a B::Buffer, Offset)>,
+}
+
+bitflags! {
+    /// https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/VkGeometryFlagBitsKHR.html
+    pub struct AccelerationStructureGeometryFlags: u32 {
+        /// indicates that this geometry does not invoke the any-hit shaders even if present in a hit group.
+        const OPAQUE = 0x1;
+        /// indicates that the implementation must only call the any-hit shader a single time for each primitive in this geometry. If this bit is absent an implementation may invoke the any-hit shader more than once for this geometry.
+        const NO_DUPLICATE_ANY_HIT_INVOCATION = 0x2;
+    }
+}
+
+#[derive(Debug)]
+pub struct AccelerationStructureGeometry<'a, B: Backend> {
+    pub flags: AccelerationStructureGeometryFlags,
+    pub geometry: Geometry<'a, B>,
+}
+
+#[derive(Debug)]
+pub enum Geometry<'a, B: Backend> {
+    Triangles(GeometryTriangles<'a, B>),
+
+    /// TODO bikeshed capitalization of AABBs.
```

Comment:
Your choice matches Rust guidelines, I think

Reply:
I have no strong opinions on naming. I'll leave as `Aabbs` but am happy to change in the future.